src/components/BuyButton.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState, Fragment } from "react";
+import { loadStripe } from "@stripe/stripe-js";
+// Make sure to call `loadStripe` outside of a component’s render to avoid
+// recreating the `Stripe` object on every render.
+const stripePromise = loadStripe(
+  process.env.REACT_APP_STRIPE_API_KEY as string
+);
+
+const BuyButton = () => {
+  const [error, setError] = useState<string>();
+
+  const handleClick = async () => {
+    // When the customer clicks on the button, redirect them to Checkout.
```

Comment:
In order to handle unexpected errors, we should wrap this whole function in a `try/catch`  block:
https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await#Adding_error_handling

Reply:
First time I've tried one of these so do correct me if I'm wrong - the Mozilla example makes sense though.